src/lib/operators/abstract_operator.cpp

```diff
@@ -14,11 +14,16 @@ AbstractOperator::AbstractOperator(const std::shared_ptr<const AbstractOperator>
                                    const std::shared_ptr<const AbstractOperator> right)
     : _input_left(left), _input_right(right) {}
 
-void AbstractOperator::execute() { _output = _on_execute(); }
+void AbstractOperator::execute() {
+  Assert(_input_left == nullptr || _input_left->_output != nullptr,
```

Comment:
Use DebugAsserts instead because this technically shouldn't go wrong in release mode?

Reply:
I feel like an `Assert` is equally valid here. These checks are really fast and only executed once per operator.